### PR TITLE
Initial Testing of new Structure

### PR DESCRIPTION
**Pull request by Woedenaz**
I am testing a new structure of the files to give us a better workflow.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "parserOptions": {
+    "sourceType": "module"
+  },
+  "extends": "eslint:recommended",
+  "rules": {
+    "quotes": ["error", "double"],
+    "semi": ["error", "always"],
+    "no-console": "off"
+  }
+}
```
Comment (ammongit) [inline .eslintrc]: Missing newline at end of file.

File: .gitignore [hunk 2]
```diff
@@ -2,11 +2,8 @@
...
 .DS_Store
-.stylelintrc
-.browserlistrc
-*.json
-modernizr-custom.js
-modernizr.js
-gulpfile.js
+
+# node items
+node_modules/
```
Comment (ammongit) [inline .gitignore]: I would move the artifacts to the top and the misc stuff to the bottom. Like:

```
# Artifacts
node_modules/
*.map

# Misc
.DS_Store
.*.swp
*.bak
```

File: gulp-tasks/browsersync.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+function reload(done) {
+  browsersync.reload();
+  done();
+}
+
+// exports
+module.exports = {
+  init: init,
+  reload: reload
+};
```
Comment (ammongit) [inline gulp-tasks/browsersync.js]: Missing newline at end of file.

File: gulp-tasks/clean.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+// Clean
+function cleanDist() {
+  return del(["./dist/"]);
+}
+
+// exports
+module.exports = {
+  dist: cleanDist
+};
```
Comment (ammongit) [inline gulp-tasks/clean.js]: Ditto for this and the rest of the files

File: gulp-tasks/copy.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+// copy assets
+function copyAssets(done) {
+  assetsDirs.forEach(dir => {
+    // src and dist
+    let sourceDir = sanitizePath(dir.src);
+    let distDir = sanitizePath(dir.dist);
+
+    // glob all files
+    let files = glob.sync(`${sourceDir}/**/*`, { nodir: true });
```
Comment (ammongit) [inline gulp-tasks/copy.js]: Isn't necessary, `somedir//a` is the same as `somedir/a`.

Comment (ammongit) [reply gulp-tasks/copy.js]: Does node not have a path class? I would feel better pushing components than interpolating a string.

File: gulp-tasks/images.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  {
+    src: "./src/assets/img/projects/*",
+    dist: "./dist/img/projects/_800x600/",
+    options: {
+      width: 800,
+      height: 600,
+      fit: "cover"
+    }
+  }
+];
```
Comment (ammongit) [inline gulp-tasks/images.js]: What's this stuff about?

File: gulp-tasks/images.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    let files = glob.sync(transform.src);
+
+    // for each file, apply transforms and save to file
+    files.forEach(function(file) {
+      let filename = path.basename(file);
+      sharp(file)
+        .resize(transform.options)
+        .toFile(`${transform.dist}/${filename}`)
+        .catch(err => {
+          console.log(err);
```
Comment (ammongit) [inline gulp-tasks/images.js]: We should be consistent about handling errors. Either throw or log.

File: gulp-tasks/scripts.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+const webpackconfig = require("../webpack.config");
+const webpackstream = require("webpack-stream");
+
+// Lint scripts
+function scriptsLint() {
+  return gulp
+    .src(
+        [
+        "./src/assets/js/modules/**/*",
+        "./gulpfile.js"
```
Comment (ammongit) [inline gulp-tasks/scripts.js]: Not indented

File: gulp-tasks/scripts.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+// Transpile, concatenate and minify scripts
+function scriptsBuild() {
+  return (
+    gulp
+      .src(
+        ["./src/assets/js/main.js"],
+        { allowEmpty: true }
+          )
```
Comment (ammongit) [inline gulp-tasks/scripts.js]: This also looks wacky.

File: package-lock.json [hunk 10]
```diff
@@ -0,0 +1,10731 @@
...
+      "dev": true
+    },
+    "chokidar": {
+      "version": "2.1.6",
+      "resolved": "https://registry.npmjs.org/chokidar/-/chokidar-2.1.6.tgz",
+      "integrity": "sha512-V2jUo67OKkc6ySiRpJrjlpJKl9kDuG+Xb8VgsGzb+aEouhgS1D0weyPU4lEzdAcsCAvrih2J2BqyXqHWvVLw5g==",
+      "dev": true,
+      "requires": {
+        "anymatch": "^2.0.0",
+        "async-
```
Comment (ammongit) [inline package-lock.json]: Can we vendor this gigantic thing

File: src/css/root.css [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+		rgba(var(--background-gradient-color),0.896) calc(0.064 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.825) calc(0.11 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.741) calc(0.166 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.648) calc(0.231 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.55) calc(0.304 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.45) calc(0.383 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.352) calc(0.467 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.259) calc(0.554 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.175) calc(0.644 * var(--background-gradient-distance)),
+		rgba(var(--background-gradient-color),0.104 )calc(0.735 * var(--background-gradient-distance)),
```
Comment (ammongit) [inline src/css/root.css]:
```suggestion
		rgba(var(--background-gradient-color),0.104) calc(0.735 * var(--background-gradient-distance)),
```

**Comment by ammongit** [comment]
Also if we're going to move to a new structure, I'd like to advocate moving "compiled" artifacts (minified stuff etc) out of the source tree and being available via github release or CI.

File: .gitignore [hunk 12]
```diff
@@ -1,9 +1,11 @@
...
-# mac items
+# Misc
 .DS_Store
+.*.swp
+*.bak
 
-# node items
-node_modules/
+#node
+package-lock.json
```
Comment (ammongit) [inline .gitignore]: Seems like your edits remove the last line. What editor are you using? I'm sure you can configure this somewhere.

**Comment by ammongit** [comment]
:tada:

**Comment by rossjrw** [comment]
I have legitmately only just now noticed that we've been using a grid structure this whole time

**Pull request merged**